src/MySqlConnector/MySqlClient/MySqlDataReader.cs

```diff
@@ -97,7 +103,20 @@ private bool ReadAsyncRemainder(PayloadData payload)
 
 		public override bool GetBoolean(int ordinal)
 		{
-			return (bool) GetValue(ordinal);
+		    object value = GetValue(ordinal);
+		    if (value is short)
+		        return (short) value != 0;
+		    if (value is ushort)
+		        return (ushort) value != 0;
+		    if (value is int)
+		        return (int) value != 0;
+		    if (value is uint)
+		        return (uint) value != 0;
+		    if (value is long)
+		        return (long) value != 0;
+		    if (value is ulong)
+		        return (ulong) value != 0;
+		    return (bool) value;
```

Comment:
I'd prefer to put an `is bool` test first, as I would expect that to be the most common  scenario.

Missing checks for `sbyte` and `byte`. Missing tests for the new behaviour.

I've pushed a commit with just this change (and new tests): 46e6bb58d565b2a392a8e7f610b6a736a457d3fa